src/Shop.js

```diff
@@ -0,0 +1,97 @@
+import { Purchase } from "./Purchase";
+import { Product } from "./Product";
+
+export class Shop {
+    constructor() {
+        this.clients = [];
+        this.products = [];
+        this.purchases = [];
+    }
+
+    addClient(client) {
+        this.clients.push(client);
+    }
+
+    deleteClient(client) {
+        this.clients.forEach((item, i, clients) => {
+            if(item.id == client.id) {
+                clients.splice(i, 1);
+            }
+        });
+    }
+
+    addProduct(product) {
+        this.products.push(product);
+    }
+
+    deleteProduct(product) {
+        this.products.forEach((item, i, products) => {
+            if(item.id == product.id) {
+                products.splice(i, 1);
+            }
+        });
+    }
+
+    addPurchase(purchase) {
+        this.purchases.push(purchase);
+    }
+
+    sellProduct(client, productName, count) {
+
+        let productsToSell = this.products.filter(item => item.name == productName),
+            infoProduct = productsToSell[0];
+
+        if(!productsToSell) {
+            return `Товара ${productName} нет на складе.`;
+        }
+        if(productsToSell.length < count) {
+            return `Товара ${productName} недостаточно на складе.`
+        }
+
+        productsToSell.splice(0, productsToSell.length - count);
+
+        let sum = count * infoProduct.cost;
+
+        productsToSell.forEach(p => {
```

Comment:
Codacy has a fix for the issue: [Expected parentheses around arrow function argument.](https://app.codacy.com/gh/Varlyle/my-first-project/pullRequest?prid=7306919)
```suggestion
        productsToSell.forEach((p) => {
```